File: LEBApp_Server/src/main/java/com/leb/app/service/dto/PointDTO.java
package com.leb.app.service.dto;

import java.io.Serializable;
import java.util.Objects;

/**
 * A DTO for the {@link com.leb.app.domain.Point} entity.
 */
public class PointDTO implements Serializable {

    private Long id;

    private String name;

    private String openingTime;

    private String closingTime;

    private String address;

    private Integer numberOfDeliveries;

    private Long ownerPoint;

    private Long status;

    private Double lat;

    private Double lng;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getOpeningTime() {
        return openingTime;
    }

    public void setOpeningTime(String openingTime) {
        this.openingTime = openingTime;
    }

    public String getClosingTime() {
        return closingTime;
    }

    public void setClosingTime(String closingTime) {
        this.closingTime = closingTime;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Integer getNumberOfDeliveries() {
        return numberOfDeliveries;
    }

    public void setNumberOfDeliveries(Integer numberOfDeliveries) {
        this.numberOfDeliveries = numberOfDeliveries;
    }

    public Long getOwnerPoint() {
        return ownerPoint;
    }

    public void setOwnerPoint(Long ownerPoint) {
        this.ownerPoint = ownerPoint;
    }

    public Long getStatus() {
        return status;
    }

    public void setStatus(Long status) {
        this.status = status;
    }

    public Double getLat() {
        return lat;
    }

    public void setLat(Double lat) {
        this.lat = lat;
    }

    public Double getLng() {
        return lng;
    }

    public void setLng(Double lng) {
        this.lng = lng;
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        result = prime * result + ((address == null) ? 0 : address.hashCode());
        result = prime * result + ((closingTime == null) ? 0 : closingTime.hashCode());
        result = prime * result + ((id == null) ? 0 : id.hashCode());
        result = prime * result + ((lat == null) ? 0 : lat.hashCode());
        result = prime * result + ((lng == null) ? 0 : lng.hashCode());
        result = prime * result + ((name == null) ? 0 : name.hashCode());
        result = prime * result + ((numberOfDeliveries == null) ? 0 : numberOfDeliveries.hashCode());
        result = prime * result + ((openingTime == null) ? 0 : openingTime.hashCode());
        result = prime * result + ((ownerPoint == null) ? 0 : ownerPoint.hashCode());
        result = prime * result + ((status == null) ? 0 : status.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        PointDTO other = (PointDTO) obj;
        if (address == null) {
            if (other.address != null)
                return false;
        } else if (!address.equals(other.address))
            return false;
        if (closingTime == null) {
            if (other.closingTime != null)
                return false;
        } else if (!closingTime.equals(other.closingTime))
            return false;
        if (id == null) {
            if (other.id != null)
                return false;
        } else if (!id.equals(other.id))
            return false;
        if (lat == null) {
            if (other.lat != null)
                return false;
        } else if (!lat.equals(other.lat))
            return false;
        if (lng == null) {
            if (other.lng != null)
                return false;
        } else if (!lng.equals(other.lng))
            return false;
        if (name == null) {
            if (other.name != null)
                return false;
        } else if (!name.equals(other.name))
            return false;
        if (numberOfDeliveries == null) {
            if (other.numberOfDeliveries != null)
                return false;
        } else if (!numberOfDeliveries.equals(other.numberOfDeliveries))
            return false;
        if (openingTime == null) {
            if (other.openingTime != null)
                return false;
        } else if (!openingTime.equals(other.openingTime))
            return false;
        if (ownerPoint == null) {
            if (other.ownerPoint != null)
                return false;
        } else if (!ownerPoint.equals(other.ownerPoint))
            return false;
        if (status == null) {
            if (other.status != null)
                return false;
        } else if (!status.equals(other.status))
            return false;
        return true;
    }

    @Override
    public String toString() {
        return "PointDTO [address=" + address + ", closingTime=" + closingTime + ", id=" + id + ", lat=" + lat
                + ", lng=" + lng + ", name=" + name + ", numberOfDeliveries=" + numberOfDeliveries + ", openingTime="
                + openingTime + ", ownerPoint=" + ownerPoint + ", status=" + status + "]";
    }

}

File: LEBApp_UI/lib/models/PointDTO.dart
import 'dart:core';
import 'package:json_annotation/json_annotation.dart';

@JsonSerializable(explicitToJson: true)
class PointDTO {

  int _id;
  String _name;
  String _openingTime;
  String _closingTime;
  String _address;
  int _numberOfDeliveries;
  int _status;
  int _ownerPoint;

  PointDTO(this._id,this._name,this._openingTime,this._closingTime,this._address,this._numberOfDeliveries,this._ownerPoint);

  int get ownerPoint => _ownerPoint;

  set ownerPoint(int value) {
    _ownerPoint = value;
  }

  int get numberOfDeliveries => _numberOfDeliveries;

  set numberOfDeliveries(int value) {
    _numberOfDeliveries = value;
  }

  String get address => _address;

  set address(String value) {
    _address = value;
  }

  String get closingTime => _closingTime;

  set closingTime(String value) {
    _closingTime = value;
  }

  String get openingTime => _openingTime;

  set openingTime(String value) {
    _openingTime = value;
  }

  int get id => _id;

  set id(int value) {
    _id = value;
  }

  int get status => _status;

  set status(int value) {
    _status = value;
  }

  String get name => _name;

  set name(String value) {
    _name = value;
  }

  PointDTO.fromJson(Map<String, dynamic> json)
      : _id = json['id'],
        _name= json['name'],
       _openingTime = json['openingTime'],
       _closingTime = json['closingTime'],
       _address = json['address'],
       _numberOfDeliveries = json['numberOfDeliveries'],
       _status = json['status'],
       _ownerPoint = json['ownerPoint'];

  Map<String, dynamic> toJson() => {
    'id' : _id,
    'name' : _name,
    'openingTime' : _openingTime,
    'closingTime' : _closingTime,
    'address' : _address,
    'numberOfDeliveries' : _numberOfDeliveries,
    'status' : _status,
    'ownerPoint' : _ownerPoint};

  @override
  String toString() {
    return 'PointDTO{_id: $_id, _name: $_name, _openingTime: $_openingTime, _closingTime: $_closingTime, _address: $_address, _numberOfDeliveries: $_numberOfDeliveries, _status: $_status, _ownerPoint: $_ownerPoint}';
  }
}

